### PR TITLE
Courses front page

### PR DESCRIPTION
**Pull request by korgan00**
## Changes

Fix https://github.com/Qiskit/qiskit.org/issues/2564

## How to read this PR

`/textbook-beta/course/machine-learning-course`
`/textbook-beta/course/introduction-course`

### CODE REVIEW

File: components/logic/CourseFrontPage.vue [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<template>
+  <main class="course-front-page">
```
Comment (techtolentino) [inline components/logic/CourseFrontPage.vue:2]: Nice, thanks for updating the class names too 👏

File: components/logic/CourseFrontPage.vue [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import { Component } from 'vue-property-decorator'
+import QiskitPage from './QiskitPage.vue'
+import { GeneralLink } from '~/constants/appLinks'
+import { Course, Prerequisite } from '~/constants/textbookBetaContent'
+
+@Component({
+  layout: 'default-max'
+})
+export default abstract class CourseFrontPage extends QiskitPage {
+  abstract routeName: string
```
Comment (techtolentino) [inline components/logic/CourseFrontPage.vue:45]: this is really cool, this `abstract`

File: components/textbook-beta/ExternalRecommendedReadings.vue [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<template>
```
Comment (techtolentino) [inline components/textbook-beta/ExternalRecommendedReadings.vue:1]: Thanks for resurrecting this!

File: pages/textbook-beta/course/introduction-course.vue [hunk 4]
```diff
@@ -1,44 +1,15 @@
...
-    return {
-      title: 'Introduction course'
-    }
-  }
+  head: () => ({
+    title: 'Introduction course'
+  })
 })
-export default class IntroductionCoursePage extends QiskitPage {
+export default class IntroductionCoursePage extends CourseFrontPage {
```
Comment (techtolentino) [inline pages/textbook-beta/course/introduction-course.vue:12]: Very nice 👏

**Review by techtolentino — DISMISSED** [review]
@korgan00 - this is great, thank you...

Question, should we also do this for the summer school pages? or do you think we save that for later?

`/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021`
`/textbook-beta/summer-school/introduction-to-quantum-computing-and-quantum-hardware-2020`

**Comment by korgan00** [comment]
> @korgan00 - this is great, thank you...
> 
> Question, should we also do this for the summer school pages? or do you think we save that for later?
> 
> `/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021` `/textbook-beta/summer-school/introduction-to-quantum-computing-and-quantum-hardware-2020`

Yeah, probably we should.

File: components/textbook-beta/ExternalRecommendedReadings.vue (hunk 3 above)
Comment (techtolentino) [inline components/textbook-beta/ExternalRecommendedReadings.vue:1]: Looks like this is handled in a new [PR](https://github.com/Qiskit/qiskit.org/pull/2566)

**Review by techtolentino — APPROVED** [review]
awesome work @korgan00 

thank you

**Pull request merged**